datasources/json/datasource.go

```diff
@@ -154,16 +156,28 @@ func (rs *RecordStream) RunWorker(ctx context.Context) error {
 		}
 
 		tx.Abort() // We only read data above, no need to risk failing now.
+		
+		if(strings.HasPrefix(rs.filePath, "http")){
+			httpClient := http.Client{
+				Timeout: time.Second * 2, // Timeout after 2 seconds
+			}
 
-		// Load/Reload file
-		file, err := os.Open(rs.filePath)
-		if err != nil {
-			return errors.Wrap(err, "couldn't open file")
+			r, err := httpClient.Get(rs.filePath)
+			if err != nil {
+				return errors.Wrap(err, "couldn't open api")
+			}
+			defer r.Body.Close()
+			rs.decoder = json.NewDecoder(r.Body)
+		} else {
```

Comment:
You're not setting the rs.file, which is getting closed in the .Close method, this will segfault.

You should still set it, and probably change the rs.file to be an io.Closer.